code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -161,15 +191,24 @@ def answer_question(self, question: str, chat_history: list[dict], **kwargs: dic
         # Generate Answer Object
         source_documents = []
         for source in sources:
-            source_document = SourceDocument(
-                id=source.metadata["id"],
-                content=source.page_content,
-                title=source.metadata["title"],
-                source=source.metadata["source"],
-                chunk=source.metadata["chunk"],
-                offset=source.metadata["offset"],
-                page_number=source.metadata["page_number"],
-            )
+            if self.env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
```

Comment:
I feel like this could be a static method on SourceDocument. e.g. `source_document = SourceDocument.Create(source=source, use_integrated_vectorisation=self.env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION)` or something similar. Interested in others views?

Reply:
Yes, if we have two flavors of SourceDocument, like we do with the AzureSearchHandler - one for IV and one for push - we can keep all the IV/non-IV handling of the source documents there. I would even move the generate_source_documents method there or under the search handler. 

Basically, QuestionAnswerTool should have as less details of what to do if AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION is true as possible.

Reply:
If we have 2 SourceDocuments, tomorrow if we add new fields, do we create another type of SourceDocument?
If we use something like `SourceDocument.Create` , won't there be a lot of code duplication because all the methods of SourceDocument will be updated?